docs/design_proposals/configurable-kubecontext.md

```diff
@@ -0,0 +1,183 @@
+  This profile is deploying to a different kubecontext `gke_abc_123`. This can be surprising to the user.
+  A solution could be to forbid specifying a kubecontext in a profile if it is activated by a kubecontext (and validate that).
+
+A natural place for the config in `skaffold.yaml` is in `latest.DeployConfig`, resulting in a json path `deploy.kubeContext`.
+
+#### Option in global Skaffold config
+Specifying a default kubecontext globally is straightforward. For example, via new config option
+```yaml
+global:
+  default-context: docker-for-desktop
+```
+
+On the other hand, building a relation between projects and kubecontext needs to solve two questions:
+
+1. How to identify projects
+2. How to save the project/kubecontext relation
+
+##### How to identify projects
+
+There are at least three possibilities:
+
+- Identify projects by their absolute host path.
+  This is guaranteed to be unique, but may break if a user moves his project to another location.
+- Identify projects by a new `metadata.name` entry in `skaffold.yaml` (see also [#2200](https://github.com/GoogleContainerTools/skaffold/issues/2200)).
+  This has the drawback of being potentially not unique, so that users accidentally pin the kubecontext for more projects than intended.
+  On the other hand, this is the standard approach taken by kubernetes resources.
+- Identify project by their initial commit.
+  This variant is stable against relocations.
+  It is also unique unless a user forks a project and wants to define different kubecontexts for each fork.
+  Compared to the other alternatives, it is rather slow.
+
+**\<What option has the best tradeoffs?\>**
+
+Resolution: We will go ahead with the `metadata.name` approach. As the name may not be unique, this requires special documentation.
+
+
+##### How to save project/kubecontext relations
+
+Currently, the Skaffold config uses the kubecontext as identifier.
+
+There are two possibilities to add the relation:
+
+- Reverse the mapping project/kubecontext and save as list under `kube-context` entries:
+  ```yaml
+  kubecontexts:
+  - kube-context: my-context
+    projects:
+    - /home/user/projects/skaffold-project
+    - project-name
+    - d76524a31c58657b06404f4442e1e64949f4d762
+  ```
+  The drawback here is that the data structure does not forbid conflicting entries, such as this:
+  ```yaml
+  kubecontexts:
+  - kube-context: context1
+    projects:
+    - my-project
+  - kube-context: context2
+    projects:
+    - my-project
+  ```
+- Add a new top-level entry in Skaffold config:
+  ```yaml
+  global: {}
+  kubecontexts: []
+  projects:
+    my-project: # project name as the key
+      kube-context: context1
+  ```
+  This option will be more complex to implement wrt `skaffold config`.
+
+**\<What Skaffold config structure has the best tradeoffs?\>**
+
+Resolution: __Not Yet Resolved__
+
+### Open Issues/Questions
+
+**\<Should there be a config option in `skaffold.yaml`?\>**
+
+Resolution: Yes.
```

Comment:
I think the overall agreement was that we do want to have the option in `skaffold.yaml`

Reply:
Yes, confirmed!